tests/unit-tests/dds/DCPS/XTypes/MemberDescriptorImpl.cpp

```diff
@@ -0,0 +1,175 @@
+  tdi->kind(TK_BOOLEAN);
+  tdi->name("Boolean");
+  DDS::TypeDescriptor_var td = tdi;
+  DynamicTypeImpl* dti = new DynamicTypeImpl();
+  DDS::DynamicType_var dt = dti;
+  dti->set_descriptor(td);
+
+  {
+    MemberDescriptorImpl md3;
+    md3.name("name");
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.id(1);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.type(dt);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.default_value("true");
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.index(2);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    DDS::UnionCaseLabelSeq label;
+    label.length(1);
+    label[0] = 3;
+    md3.label(label);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.try_construct_kind(DDS::USE_DEFAULT);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.is_key(true);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.is_optional(true);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.is_must_understand(true);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.is_shared(true);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+  {
+    MemberDescriptorImpl md3;
+    md3.is_default_label(true);
+    EXPECT_TRUE(md3.equals(&md3));
+    EXPECT_FALSE(md3.equals(&md1));
+    EXPECT_FALSE(md1.equals(&md3));
+  }
+}
+
+TEST(dds_DCPS_XTypes_MemberDescriptorImpl, MemberDescriptorImpl_copy_from)
+{
+  TypeDescriptorImpl* tdi = new TypeDescriptorImpl();
+  tdi->kind(TK_BOOLEAN);
+  tdi->name("Boolean");
+  DDS::TypeDescriptor_var td = tdi;
+  DynamicTypeImpl* dti = new DynamicTypeImpl();
+  DDS::DynamicType_var dt = dti;
+  dti->set_descriptor(td);
+
+  MemberDescriptorImpl md1;
+  EXPECT_EQ(md1.copy_from(0), DDS::RETCODE_BAD_PARAMETER);
+
+  MemberDescriptorImpl md2;
+  md2.name("name");
+  md2.id(1);
+  md2.type(dt);
+  md2.default_value("true");
```

Comment:
Is this supposed to be `TRUE` or does it not matter in this context?

Reply:
Yes.